src/core/public/application/application_service.tsx

```diff
@@ -105,23 +153,41 @@ export class ApplicationService {
     }
 
     // Disable registration of new applications
-    this.apps$.complete();
-    this.legacyApps$.complete();
+    this.started = true;
 
     const legacyMode = injectedMetadata.getLegacyMode();
     const currentAppId$ = new BehaviorSubject<string | undefined>(undefined);
-    const { availableApps, availableLegacyApps, capabilities } = await this.capabilities.start({
-      apps: new Map([...this.apps$.value].map(([id, { app }]) => [id, app])),
-      legacyApps: this.legacyApps$.value,
+    const { availableApps, capabilities } = await this.capabilities.start({
+      apps: new Map(
+        [...this.apps].map(([id, appBox]) => [
+          id,
+          (isLegacyApp(appBox) ? appBox : appBox.app) as App | LegacyApp,
+        ])
+      ),
```

Comment:
Just a heads up, this will probably create conflicts in https://github.com/elastic/kibana/pull/51527

Reply:
Yea, well aware of that. We need to see with @eliperelman how we synchronize these PRs